src/engine/sparqlExpressions/SparqlExpressionTypes.h

```diff
@@ -129,6 +130,20 @@ inline ExpressionResult copyExpressionResultIfNotVector(
   return std::visit(copyIfCopyable, result);
 }
 
+// TODO: Docs
+inline ExpressionResult copyExpressionResult(const ExpressionResult& result) {
+  auto copyIfCopyable =
+      []<SingleExpressionResult R>(const R& x) -> ExpressionResult {
+    if constexpr (std::is_copy_assignable_v<R> &&
+                  std::is_copy_constructible_v<R>) {
+      return x;
+    } else {
+      return x.clone();
+    }
+  };
+  return std::visit(copyIfCopyable, result);
```

Comment:
```suggestion
  return std::visit(copyIfCopyable, AD_FWD(result));
```

Those two changes should move the result IF it is movable (e.g. because you moved it in). 
You can write a test for this behavior (e.g. if you pass in a vector by moving, it should be empty afterwards).